### PR TITLE
Add LLVM toolchain

### PR DESCRIPTION
**Pull request by siddharthab**
New module toolchains_llvm@0.10.3.

### CODE REVIEW

File: modules/toolchains_llvm/0.10.1/MODULE.bazel [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+module(
+    name = "toolchains_llvm",
+    version = "0.10.0",
```
Comment (jsharpe) [inline modules/toolchains_llvm/0.10.1/MODULE.bazel:17]: This should be 0.10.1

Comment (siddharthab) [reply modules/toolchains_llvm/0.10.1/MODULE.bazel]: Done. Thanks. I was not sure if I also needed to patch the version in the original source archive.

Comment (siddharthab) [reply modules/toolchains_llvm/0.10.1/MODULE.bazel]: Looks like I do need to patch.

Comment (siddharthab) [reply modules/toolchains_llvm/0.10.1/MODULE.bazel]: I changed my release process to automatically change this version string.

**Comment by siddharthab** [comment]
All tests are now passing.

File: modules/toolchains_llvm/0.10.2/MODULE.bazel [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+module(
+    name = "toolchains_llvm",
+    version = "0.10.2",
+    compatibility_level = 0,
+)
+
+bazel_dep(name = "bazel_skylib", version = "1.4.2")
```
Comment (dzbarsky) [inline modules/toolchains_llvm/0.10.2/MODULE.bazel:21]: @siddharthab should skylib have `dev_dependency = True` set?

Comment (siddharthab) [reply modules/toolchains_llvm/0.10.2/MODULE.bazel:21]: We do make the user use `native_binary` rule from skylib, so unfortunately, this is a real dependency.

**Review by dzbarsky — APPROVED** [review]

**Comment by fmeum** [comment]
@meteorcloudy @Wyverald I personally agree with the somewhat generic name as this really is _the_ LLVM Bazel toolchain, just want to confirm this with you before I hit merge.

**Review by meteorcloudy — APPROVED** [review]
LGTM, thanks!

**Pull request merged**